Sources/JWTKit/JWTParser.swift

```diff
@@ -42,3 +34,27 @@ struct JWTParser: Sendable {
         encodedHeader + [.period] + encodedPayload
     }
 }
+
+public struct DefaultJWTParser: JWTParser {
```

Comment:
Given this is what will be used, I'm not a massive fan of the API, as it might get confusing. We could rename it to the type as `JWTParser` and make the protocol `CustomJWTParser` but I'm not sure that's better